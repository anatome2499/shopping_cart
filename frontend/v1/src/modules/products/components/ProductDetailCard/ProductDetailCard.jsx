import React from 'react'
import classNames from 'classnames'
import { Paper, Grid, Button } from '@material-ui/core'

const ProductCard = ({
  imgUrl,
  id,
  name,
  gender,
  age,
  price,
  availability,
  handleOnClick,
  classes
}) => {
  const onClick = () => {
    handleOnClick(id)
  }
  return (
    <Paper>
      <Grid container spacing={1}>
        <Grid item xs={4}>
          <Grid item sm={12} >
            <img src={imgUrl} className={classes.image}/>
          </Grid>
        </Grid>
        <Grid item xs={4}>
          <p>Name: {name}</p>
          <p>Gender: {gender}</p>
          <p>Age: {age}</p>
          <p>${price}</p>
          <p>{availability}</p>
        </Grid>
        <Grid item xs={4} className={classes.center}>
          <Button onClick={onClick} className={classes.button}>
            Add to Cart
          </Button>
        </Grid>
      </Grid>
    </Paper>
  )
}

export default ProductCard
