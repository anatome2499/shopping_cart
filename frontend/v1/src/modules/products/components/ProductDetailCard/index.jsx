import React from 'react'
import styles from './styles'
import { withStyles, Paper, Grid, Button } from '@material-ui/core'

const ProductCard = () => {
    return (
        <Paper>
            <Grid container spacing={1}>
              <Grid item xs={4}></Grid>
              <Grid item xs={4}></Grid>
              <Grid item xs={4}>
                <Button>
                  Add to Cart
                </Button>
              </Grid>
            </Grid>
        </Paper>
    )
}

export default withStyles(styles) (ProductCard)
