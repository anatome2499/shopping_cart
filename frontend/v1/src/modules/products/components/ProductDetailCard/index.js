import { withStyles } from '@material-ui/core'

import styles from './styles'
import ProductDetailCard from './ProductDetailCard'

export default withStyles(styles)(ProductDetailCard)