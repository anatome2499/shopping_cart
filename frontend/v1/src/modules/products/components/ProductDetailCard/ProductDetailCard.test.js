import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Button } from '@material-ui/core'
import ProductDetailCard from './ProductDetailCard'

describe('<ProductCard />', () => {
  let component
  const handleOnClick = jest.fn()
  beforeEach(() => {
    component = shallow(<ProductDetailCard classes={{}}/>)

  })

  it('should render with no errors', () => {
    
    component.setProps({
      imgUrl: 'https://s3-ap-southeast-1.amazonaws.com/assets-eng-prod/products/th_310093-3.jpg',
      name: 'Princess Palace',
      gender: 'Female',
      age: '3_to_5',
      price: 24.95,
      availability: 'In Stock',
      handleOnClick: handleOnClick
    })
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should call function when button onClick', () => {
    component.setProps({
      handleOnClick: handleOnClick
    })
    const button = component.find(Button)

    button.simulate('click')
    expect(handleOnClick).toHaveBeenCalled()
  })
})
