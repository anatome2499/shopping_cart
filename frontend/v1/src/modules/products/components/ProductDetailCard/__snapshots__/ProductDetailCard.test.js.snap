// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should render with no errors 1`] = `
<WithStyles(ForwardRef(Paper))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={1}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={12}
      >
        <img
          src="https://s3-ap-southeast-1.amazonaws.com/assets-eng-prod/products/th_310093-3.jpg"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={4}
    >
      <p>
        Name: 
        Princess Palace
      </p>
      <p>
        Gender: 
        Female
      </p>
      <p>
        Age: 
        3_to_5
      </p>
      <p>
        $
        24.95
      </p>
      <p>
        In Stock
      </p>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={4}
    >
      <WithStyles(ForwardRef(Button))
        onClick={[Function]}
      >
        Add to Cart
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Paper))>
`;
